HeterogeneousCore/CUDACore/src/chooseDevice.cc

```diff
@@ -1,11 +1,20 @@
 #include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Utilities/interface/Exception.h"
 #include "HeterogeneousCore/CUDAServices/interface/CUDAService.h"
 
 #include "chooseDevice.h"
 
 namespace cms::cuda {
   int chooseDevice(edm::StreamID id) {
     edm::Service<CUDAService> cudaService;
+    if (not cudaService->enabled()) {
```

Comment:
Should this be
```suggestion
    if (not cudaService or not cudaService->enabled()) {
```
to catch the case where the `CUDAService` is not included in the process ?

Reply:
De-referencing the `edm::Service<CUDAService>` without `CUDAService` in the configuration gives the standard exception
```
Service Request unable to find requested service with compiler type name ' 11CUDAService'.
```
I thought that would be sufficient to cover that case.